src/firebolt/async_db/cursor.py

```diff
@@ -100,11 +106,11 @@ async def _raise_if_error(self, resp: Response) -> None:
                 f"Error executing query:\n{resp.read().decode('utf-8')}"
             )
         if resp.status_code == codes.FORBIDDEN:
-            if self.connection.database and not await self.is_db_available(
-                self.connection.database
+            if self.parameters["database"] and not await self.is_db_available(
```

Comment:
I think we should better use `self.database` here. Also, I believe that if "database" is not in parameter it will fail with an exception.

Reply:
Here and in other places